twoliter/src/cargo_make.rs

```diff
@@ -0,0 +1,213 @@
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use log::trace;
+use tokio::process::Command;
+
+use crate::common::exec;
+use crate::docker::ImageArchUri;
+use crate::project::Project;
+
+fn require_sdk(project: &Project, arch: &str) -> Result<(ImageArchUri, ImageArchUri)> {
+    match (project.sdk(arch), project.toolchain(arch)) {
+        (Some(s), Some(t)) => Ok((s, t)),
+        _ => bail!(
+            "When using twoliter make, it is required that the SDK and toolchain be specified in \
+            Twoliter.toml"
+        ),
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct CargoMake {
+    makefile_path: Option<PathBuf>,
+    project_dir: Option<PathBuf>,
+    args: Vec<String>,
+}
+
+impl CargoMake {
+    pub fn new<S>(project: &Project, arch: S) -> Result<Self>
+    where
+        S: Into<String>,
+    {
+        let (sdk, toolchain) = require_sdk(project, &arch.into())?;
+        Ok(Self::default()
+            .env("TLPRIVATE_SDK_IMAGE", sdk)
+            .env("TLPRIVATE_TOOLCHAIN", toolchain))
+    }
+
+    pub fn makefile<P>(mut self, makefile_path: P) -> Self
```

Comment:
Acknowledging that you may have copied this from me...  In case we do make a lib at some point, all things should start off as pub(crate). Then we would make certain things pub only if a lib were introduced.

```suggestion
    pub(crate) fn makefile<P>(mut self, makefile_path: P) -> Self
```